crypto/bio/bio_lib.c

```diff
@@ -47,25 +51,33 @@ static long bio_call_callback(BIO *b, int oper, const char *argp, size_t len,
      */
     if (HAS_LEN_OPER(bareoper)) {
         /* In this case |len| is set, and should be used instead of |argi| */
-        if (len > INT_MAX)
-            return -1;
+        if (len > INT_MAX) {
+            ret = -1;
+            goto end;
+        }
 
         argi = (int)len;
 
         if (inret && (oper & BIO_CB_RETURN)) {
-            if (*processed > INT_MAX)
-                return -1;
-            inret = *processed;
+            if (int_processed > INT_MAX) {
+                ret = -1;
+                goto end;
+            }
+            inret = int_processed;
         }
     }
 
     ret = b->callback(b, oper, argp, argi, argl, inret);
 
     if (ret >= 0 && (HAS_LEN_OPER(bareoper) || bareoper == BIO_CB_PUTS)) {
-        *processed = (size_t)ret;
```

Comment:
I think I must be missing the point of this PR. This is the only place I can see a potential NULL deref? Or have I missed something? If it is only here why isn't it sufficient just to do `if (processed != NULL) *processed = (size_t)ret`